### PR TITLE
[API v3] add CRON_SAFE_MODE

### PR DESCRIPTION
**Pull request by Alys**
Partial work for https://github.com/HabitRPG/habitrpg/issues/7161
### Changes
- Adds "safe mode" for cron.
- Requires a **new environment variable**: `CRON_SAFE_MODE` (set to "false" normally; "true" when we want safe mode).
- Removes an unrelated TODO that's already been done.
- Removes duplicated code (this was causing twice as much mana as appropriate to be gained overnight in API v3 only, so if anyone reports that as a bug, this fixes it).

I suggest reviewing with `w=1` because a bunch of unchanged code is now wrapped in an `else`.

The quest damage and chat message portions still need to be tested (can't start a quest due to an unrelated bug).

I'm writing automated tests for this.

### CODE REVIEW

File: website/server/models/group.js [hunk 1]
```diff
@@ -515,8 +515,10 @@ schema.statics.bossQuest = async function bossQuest (user, progress) {
   let down = progress.down * quest.boss.str; // multiply by boss strength
 
   group.quest.progress.hp -= progress.up;
-  // TODO Create a party preferred language option so emits like this can be localized
-  group.sendChat(`\`${user.profile.name} attacks ${quest.boss.name('en')} for ${progress.up.toFixed(1)} damage.\` \`${quest.boss.name('en')} attacks party for ${Math.abs(down).toFixed(1)} damage.\``);
+  // TODO Create a party preferred language option so emits like this can be localized. Suggestion: Always display the English version too. Or, if English is not displayed to the players, at least include it in a new field in the chat object that's visible in the database - essential for admins when troubleshooting quests!
+  let playerAttack = `${user.profile.name} attacks ${quest.boss.name('en')} for ${progress.up.toFixed(1)} damage.`;
+  let bossAttack = nconf.get('CRON_SAFE_MODE') ? `${quest.boss.name('en')} did not attack the party because it was asleep while maintenance was happening.` : `${quest.boss.name('en')} attacks party for ${Math.abs(down).toFixed(1)} damage.`;
```
Comment (paglias) [inline website/server/models/group.js]: nconf.get('CRON_SAFE_MODE') === 'true'

File: website/server/libs/api-v3/cron.js [hunk 2]
```diff
@@ -231,14 +239,9 @@ export function cron (options = {}) {
...
   // Adjust for fraction of dailies completed
-  user.stats.mp += _.max([10, 0.1 * user._statsComputed.maxMP]) * dailyChecked / (dailyDueUnchecked + dailyChecked);
-  if (user.stats.mp > user._statsComputed.maxMP) user.stats.mp = user._statsComputed.maxMP;
-
   if (dailyDueUnchecked === 0 && dailyChecked === 0) dailyChecked = 1;
   user.stats.mp += _.max([10, 0.1 * user._statsComputed.maxMP]) * dailyChecked / (dailyDueUnchecked + dailyChecked);
-  if (user.stats.mp > user._statsComputed.maxMP) {
-    user.stats.mp = user._statsComputed.maxMP;
-  }
+  if (user.stats.mp > user._statsComputed.maxMP) user.stats.mp = user._statsComputed.maxMP;
```
Comment (paglias) [inline website/server/libs/api-v3/cron.js:244]: 👍

**Comment by paglias** [comment]
Looking good!

File: website/server/models/group.js (hunk 1 above)
Comment (Alys) [inline website/server/models/group.js]: Ugh, good point, thanks. Fixed. This is ready for review again.

**Pull request merged**